src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppCertificate.cs

```diff
@@ -0,0 +1,121 @@
+using Microsoft.Azure.Commands.WebApps.Models.WebApp;
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using Microsoft.Azure.Management.WebSites.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Linq;
+using System.Management.Automation;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.Certificates
+{
+
+    /// <summary>
+    /// This commandlet will let you create a new managed certificate
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "WebAppCertificate", SupportsShouldProcess = true), OutputType(typeof(PSCertificate))]
+    public class NewAzureWebAppCertificate : WebAppBaseClientCmdLet
+    {
+        const string CertNamePostFixSeparator = "_";
+        const string ParameterSet1Name = "S1";
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 0, Mandatory = true, HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 1, Mandatory = true, HelpMessage = "The name of the web app.")]
+        [ResourceNameCompleter("Microsoft.Web/sites", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string WebAppName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 2, Mandatory = false, HelpMessage = "The name of the web app slot.")]
+        [ResourceNameCompleter("Microsoft.Web/sites/slots", "ResourceGroupName", "WebAppName")]
+        [ValidateNotNullOrEmpty]
+        public string Slot { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 3, Mandatory = true, HelpMessage = "Custom hostnames associated with web app/slot.")]
+        [ValidateNotNullOrEmpty]
+        public string HostName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Mandatory = false, HelpMessage = "To add the created certificate to WebApp/slot.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter AddBinding { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSet1Name, Mandatory = false, HelpMessage = "Ssl state option. Use either 'SniEnabled' or 'IpBasedEnabled'. Default option is 'SniEnabled'.")]
+        [ValidateNotNullOrEmpty]
+        public SslState? SslState { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (!string.IsNullOrWhiteSpace(ResourceGroupName) && !string.IsNullOrWhiteSpace(WebAppName))
+            {
+                //PSSite webApp = null;
+                var webApp = new PSSite(WebsitesClient.GetWebApp(ResourceGroupName, WebAppName, Slot));
+                var location = webApp.Location;
+
+                Certificate createdCertdetails = null;
+
+                var certificate = new Certificate(
```

Comment:
Add a name here? so that the canonical Name is not used as the cert Name, Ex of Cert Object, also pass the validation method. 

{
      "name": "<certNamehere>",
      "type": "Microsoft.Web/certificates",
      "location": "<>",
      "properties": {
      "serverFarmId": "<>",
      "canonicalName": "<>,
      "domainValidationMethod":"http-token"
        
      },
}

Reply:
added 'type' but don't have property for **domainValidationMethod** in certificate.cs